iUAEParents/MainEmulationViewController.mm

```diff
@@ -126,12 +126,65 @@ - (void)viewDidLoad {
 
 }
 
+extern void set_MainView(MainEmulationViewController *m);
 - (void)viewDidAppear:(BOOL)animated {
     [super viewDidAppear:animated];
     [self applyConfiguredEffect];
     set_joystickactive();
+    set_MainView(self);
+    
     [_mouseHandler reloadMouseSettings];
     [_joyController reloadJoypadSettings];
+    
+    
+    if(mainMenu_servermode != 1)
+    {
+        if(advertiser!=nil)
+        {//stop server
+            [self stopServer];
+        }
+        else if(session != nil && mainMenu_servermode == 0)
+        {//close client connection
+            [session disconnect];
+            session = nil;
+            [self showMessage: @"closed connection" withMessage: @"to the server"];
+            
+        }
+        //the device should go to sleep after some idle time
+        [[UIApplication sharedApplication] setIdleTimerDisabled:NO];
+    }
+    
+    if(mainMenu_servermode == 1)
```

Comment:
Minor thing: it would help with code readability to use constants instead of hardcoded integer literals, so 

if (server mode == kServerEnabled) ... or such.